WordPress/src/main/java/org/wordpress/android/ui/posts/PostListFragment.kt

```diff
@@ -204,100 +267,46 @@ class PostListFragment : Fragment(),
-    }
-
-    override fun onResume() {
-        super.onResume()
-
-        // always (re) load when resumed to reflect changes made elsewhere
-        loadPosts(LoadMode.IF_CHANGED)
-
-        // scale in the fab after a brief delay if it's not already showing
-        if (fabView?.visibility != View.VISIBLE) {
-            val delayMs = resources.getInteger(R.integer.fab_animation_delay).toLong()
-            Handler().postDelayed({
-                if (isAdded) {
-                    AniUtils.scaleIn(fabView, AniUtils.Duration.MEDIUM)
-                }
-            }, delayMs)
-        }
-    }
-
-    private fun requestPosts(loadMore: Boolean) {
-        if (!isAdded || isFetchingPosts) {
-            return
-        }
-
         if (!NetworkUtils.isNetworkAvailable(nonNullActivity)) {
+            swipeRefreshLayout?.isRefreshing = false
             updateEmptyView(EmptyViewMessageType.NETWORK_ERROR)
-            return
-        }
-
-        if (postListAdapter.itemCount == 0) {
-            updateEmptyView(EmptyViewMessageType.LOADING)
-        }
-
-        isFetchingPosts = true
-        if (loadMore) {
-            showLoadMoreProgress()
+            // If network is not available, we can refresh the items from the DB in case an update is not reflected
+            // It really shouldn't be necessary, but wouldn't hurt to have it here either
+            refreshListManagerFromStore(listDescriptor, shouldRefreshFirstPageAfterLoading = false)
+        } else {
+            listManager?.refresh()
         }
-
-        val payload = FetchPostsPayload(site, loadMore)
-        dispatcher.dispatch(PostActionBuilder.newFetchPostsAction(payload))
-    }
-
-    private fun showLoadMoreProgress() {
-        progressLoadMore?.visibility = View.VISIBLE
-    }
-
-    private fun hideLoadMoreProgress() {
-        progressLoadMore?.visibility = View.GONE
     }
 
-    /*
-     * Upload started, reload so correct status on uploading post appears
-     */
-    fun onEventMainThread(event: PostEvents.PostUploadStarted) {
-        if (isAdded && site.id == event.mLocalBlogId) {
-            loadPosts(LoadMode.FORCED)
+    private fun newPost() {
+        if (!isAdded) {
+            return
         }
+        ActivityLauncher.addNewPostForResult(nonNullActivity, site, false)
     }
 
-    /*
-     * Upload cancelled (probably due to failed media), reload so correct status on uploading post appears
-     */
-    fun onEventMainThread(event: PostEvents.PostUploadCanceled) {
-        if (isAdded && site.id == event.localSiteId) {
-            loadPosts(LoadMode.FORCED)
+    private fun updateEmptyViewForListManagerChange(listManager: ListManager<PostModel>) {
+        if (!listManager.isFetchingFirstPage) {
+            if (listManager.size == 0) {
+                if (NetworkUtils.isNetworkAvailable(nonNullActivity)) {
+                    updateEmptyView(EmptyViewMessageType.NO_CONTENT)
+                }
```

Comment:
This is actually a small workaround I added, because the empty view would briefly show the no content view when we are actually fetching the first page. Having said that, I am not even 100% sure about that claim, as I might be getting confused in a sea of issues I dealt with. This certainly requires at least a comment or an improvement before this PR is merged in.

Great catch, thanks @malinajirka!

Reply:
@malinajirka I've improved the empty view handling in ee5e054544a044896ebbbbd749b2ccc49656a516. It should be much clearer now.

It's still not perfect though. I think ideally we'd want the `ListManager` to carry that information and the empty view is updated only through `ListManager` changes. I do have some plans for that, but not right now.